mlflow/models/model_config.py

```diff
@@ -11,8 +11,73 @@
 
 class ModelConfig:
     """
-    ModelConfig used in code to read a YAML configuration file, and this configuration file can be
-    overridden when logging a model.
+    ModelConfig used in code to read a YAML configuration file or a dictionary.
+    When invoking the ModelConfig locally in a model file, development_config can be passed in
+    which would be used as configuration for the model. But this development_config configuration
+    file will be overridden when logging a model. When no model_config is passed in while logging
+    the model, an error will be raised when trying to load the model using ModelConfig.
+    Note: development_config is not used when logging the model.
```

Comment:
Can we explain how `development_config` works using the code examples?

Reply:
Doesn't the last example help with that? Let me know if I am missing something and like to explain something else better?

Reply:
It would be nice to get a doc team member's review for clarity, maybe Peter Lu (can't tag him here)? Agree that the role of ModelConfig could be clarified here, basically it's a way to parametrize code in a custom pyfunc that's logged using model-from-code, with defaults defined by `development_config` etc

Reply:
Also seems like the behavior is that any passed-in config takes precedent over default `development_config`, i.e. they're not merged (it's a little different from e.g. python kwargs) - it would also be good to clarify this

Reply:
@sunishsheth2009 

I'd prefer something like this:

```
When invoking the ModelConfig locally in a model file, development_config can be passed in
    which would be used as configuration for the model. But this development_config configuration
    file will be overridden when logging a model. Here's an example:

<Example>

When no model_config is passed in while logging
    the model, an error will be raised when trying to load the model using ModelConfig. Here's an example:

<Example>
```

Reply:
@sunishsheth2009 

> Doesn't the last example help with that? Let me know if I am missing something and like to explain something else better?

Sorry, I wasn't clear. I would reorganize the docstring like this for clarity:

```
When invoking the ModelConfig locally in a model file, development_config can be passed in which would be used as configuration for the model:

<Code example>

But this development_config configuration file will be overridden when logging a model. When no model_config is passed in while logging the model:

<Code example>

When no model_config is passed in while logging the model, an error will be raised when trying to load the model using ModelConfig.

<Code example>
```